arduino/libraries/librariesmanager/install.go

```diff
@@ -111,21 +112,42 @@ func (lm *LibrariesManager) InstallZipLib(ctx context.Context, archivePath strin
 }
 
 //InstallGitLib  installs a library hosted on a git repository on the specified path.
-func (lm *LibrariesManager) InstallGitLib(url string) error {
+func (lm *LibrariesManager) InstallGitLib(gitURL string) error {
 	libsDir := lm.getUserLibrariesDir()
 	if libsDir == nil {
 		return fmt.Errorf("User directory not set")
 	}
-	i := strings.LastIndex(url, "/")
-	folder := strings.TrimRight(url[i+1:], ".git")
-	path := libsDir.Join(folder)
 
-	_, err := git.PlainClone(path.String(), false, &git.CloneOptions{
-		URL:      url,
+	libraryName, err := parseGitURL(gitURL)
+	if err != nil {
+		return err
+	}
+
+	installPath := libsDir.Join(libraryName)
+
+	_, err = git.PlainClone(installPath.String(), false, &git.CloneOptions{
+		URL:      gitURL,
 		Progress: os.Stdout,
 	})
 	if err != nil {
 		return err
 	}
 	return nil
 }
+
+func parseGitURL(gitURL string) (string, error) {
+	var res string
+	if strings.HasPrefix(gitURL, "git") || strings.HasPrefix(gitURL, "ssh") {
+		// We can't parse these as URLs
```

Comment:
Is this for the `git://github.com/foo/bar` and `ssh://git@github.com/foo/bar.git` style of URLs? `url.Parse()` seem to handle them fine in my quick experiments.

Reply:
Nope, this is about URLs like `git@github.com:arduino/arduino-cli.git`. I could probably change this to `strings.HasPrefix(gitURL, "git@")`. :thinking: 

I'll give it a try.

Reply:
Thanks. I think that's a nice improvement.